AutomatedTesting/Gem/PythonTests/Atom/atom_utils/screenshot_utils.py

```diff
@@ -91,19 +92,23 @@ def wait_until_screenshot(self):
                 frames_waited = frames_waited + 1
         general.log(f"(waited {frames_waited} frames)")
 
-
 def take_screenshot_game_mode(screenshot_name, entity_name=None):
     """
     Enters game mode & takes a screenshot, then exits game mode after.
-    :param screenshot_name: name to give the captured screenshot .ppm file.
+    :param screenshot_name: name to give the captured screenshot .png file.
     :param entity_name: name of the entity being tested (for generating unique log lines).
     :return: None
     """
     general.enter_game_mode()
     helper.wait_for_condition(lambda: general.is_in_game_mode(), 2.0)
     general.log(f"{entity_name}_test: Entered game mode: {general.is_in_game_mode()}")
-    ScreenshotHelper(general.idle_wait_frames).capture_screenshot_blocking(f"{screenshot_name}.ppm")
+    ScreenshotHelper(general.idle_wait_frames).capture_screenshot_blocking(f"{screenshot_name}.png")
     general.idle_wait(1.0)
     general.exit_game_mode()
     helper.wait_for_condition(lambda: not general.is_in_game_mode(), 2.0)
     general.log(f"{entity_name}_test: Exit game mode: {not general.is_in_game_mode()}")
+
+def compare_screenshots(imageA, imageB):
+    result = azlmbr.atom.FrameCaptureTestRequestBus(
+        azlmbr.bus.Broadcast, "CompareScreenshots", imageA, imageB, 0.01)
+    return result
```

Comment:
before you return result you should assert result_code == ImageDiffResultCode::Success however you would represent that in the code here. that depends on what result is in this case but it seems to be an ebus outcome which might have IsSuccess() so you could assert result.IsSuccess()

Reply:
This is actually a good suggestion. But I think if we want to change the comparison part, we have to change the whole `FrameCaptureSystemComponent` to match the same style. It contains `CaptureScreenshot` that returns a `FrameCaptureId`, and `BuildScreenshotFilePath` that returns a `string`. Those can be all put into outcome based on their usage.
It will expand the scope of this PR. I'll open a ticket to refactor this.